### ISSUE TITLE
autobind-decorator should be a runtime dependency

### ISSUE THREAD

**Issue by laviddichterman** [VilledeMontreal/express-idempotency#10]
When I follow the example code here: https://github.com/VilledeMontreal/express-idempotency/blob/master/examples/simple-use/src/index.js in my own application, I get the following stack trace:

`internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'autobind-decorator'
Require stack:
- /home/lavid/Documents/wcp-order-backend/node_modules/express-idempotency/dist/services/idempotencyService.js
- /home/lavid/Documents/wcp-order-backend/node_modules/express-idempotency/dist/index.js
- /home/lavid/Documents/wcp-order-backend/server.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
`

It seems that autobind-decorator should be a runtime dependency? https://github.com/VilledeMontreal/express-idempotency/blob/9922b0cd3521ae4b7ed4d3f8754cbae7c8ce135d/package.json#L40

**Comment by joscelynjean** [comment]
Hi, thank you for your comment.

I will have a look at this. I thought that ```autobind-decorator``` was only use by the transpiler.

Just to help with my investigation, can you provide un snippet of your code using the idempotency service?

**Comment by laviddichterman** [comment]
Thanks for the quick response. I think that depending on how the package is used, it can be used to transpile or be used during runtime. In the express-idempotency/dist/services/idempotencyService.js file in your package, it still has `const autobind_decorator_1 = require("autobind-decorator");` so it would be runtime in this case.

As requested, here's how I'm using it (to be fair, I'm not even using it yet, just added the middleware at this point).
```const express = require('express');
const http = require("http");
const socketIo = require("socket.io");
const bodyParser = require('body-parser');
const moment = require('moment');
const cors = require('cors');
const logger = require("./logging");
const app = express();
const idempotency = require('express-idempotency');
const expressWinston = require('express-winston')
const router = require('./routes/')()
...
app.use(idempotency.idempotency());
```

**Comment by YouvaB** [comment]
@laviddichterman As a workaround, can you try to add `autobind-decorator` in your package.json? We'll change the package.json of the library, but in the meantime you might be able to make it work that way.

**Comment by laviddichterman** [comment]
@YouvaB yes, that works just fine. I just wanted to bring the dependency issue to your attention.

**Comment by YouvaB** [comment]
@laviddichterman thank you very much, that's really appreciated!

**Comment by joscelynjean** [comment]
@laviddichterman the fix is now available in 1.0.3. Thank you for your feedback.
